src/assign.rs

```diff
@@ -58,25 +54,456 @@ pub struct Assignment<'a> {
     ///
     /// Note: if a portion of the path contains a leaf node that is to be
     /// overridden by an object or an array, then the path will be leaf that is
-    /// replaced.
-    ///
-    /// For example, if you had the json:
-    /// ```json
-    /// { "foo:" "bar" }
-    /// ```
-    /// and you assigned `"new_value"` to `"/foo/bar/baz"`, then the value would
-    /// be `Some("/foo/bar")`.
-    pub created_or_mutated: PointerBuf,
-    /// A `Pointer` consisting of the path which was assigned.
+    /// replaced. For example, given the json `{ "foo:" "bar" }`, if `"new_value"` is
+    /// assigned to `"/foo/bar/baz"`, then `assigned_to` would be `/foo/bar"`
     ///
     /// ## Example
     /// ```rust
-    /// use serde_json::json;
-    /// use jsonptr::{Pointer, Assign};
+    /// # use serde_json::json;
+    /// # use jsonptr::{Pointer, Assign};
     /// let mut data = json!({ "foo": ["zero"] });
     /// let mut ptr = Pointer::from_static("/foo/-");
     /// let assignment = data.assign(&mut ptr, "one").unwrap();
     /// assert_eq!(assignment.assigned_to, Pointer::from_static("/foo/1"));
     /// ```
     pub assigned_to: PointerBuf,
```

Comment:
I don't think this is worth changing in this MR, but we should consider if the cost of keeping track of the assignment path is worth it, since it adds extra allocations. Could maybe be an opt-in thing, either through a flag or separate method.

Reply:
So I fixed `assigned_to` in this PR. It now reports appropriately the path of `assigned` (mut ref to the src assigned). Before, it was only being updated if `"-"` existed in the path, basically being the full pointer. Now it's wherever the assignment ends.

If we keep auto-expansion of any sort, this probably needs to stay. However, it could be made into a `Cow<'p, Pointer>`. It'll take more work / logic but in most cases, it won't allocate.

Reply:
I haven't used this API either but it seems like someone may want to know what ended up getting assigned and where. I may switch it over to `Cow<'p, Pointer>` even if it is always `Cow::Owned` for now.

Reply:
im removing it. its akward when it comes to replacement of key,

```rust
let data = json!{ "foo": "str"};
let assignment = data.assign(Pointer::from_static("/foo/bar/baz", json!("qux")).unwrap();
// should this be assigned to `"foo"` or `"foo/bar"`? 
assert_eq!(assignment.assigned_to, ???);
assert_eq!(assignment.assigned, json!({"bar": {"baz": "qux"}}))
```